protocol/0057-TRAN-transfers.md

```diff
@@ -109,6 +109,7 @@ A party should be able to configure the distribution of rewards by specifying th
 
 - `window_length` - the number of epochs over which to evaluate the reward metric. The value should be limited to 100 epochs.
 - `lock_period` - the number of epochs after distribution to delay [vesting of rewards](./0085-RVST-rewards_vesting.md#vesting-mechanics) by.
+- `cap_reward_fee_multiple` [optional] - if set, the actual amount of reward transferred to each public key during distribution for this transfer will be `min(calculated_reward, cap_reward_fee_multiple × fees_paid_this_epoch)`. If all keys are capped (i.e. the total amount of the transfer cannot be be sent to eligible keys without breaching the cap) then the remaining balance must either not leave the source account for an iteration of the recurring transfer, or be transferred back to the source account with no fee. If this occurs, and the total transferred in a given epoch this does not affect the size of the next iteration, which proceeds as normal (including decay factors etc.) as if the full transfer has been made.
```

Comment:
You need to use the quantum values for both assets involved and create a conversion factor.

Reply:
> If all keys are capped (i.e. the total amount of the transfer cannot be be sent to eligible keys without breaching the cap) 

Isn't that "if at least one of the keys got capped"? Because that is the condition to have leftover reward balance.

Reply:
> then the remaining balance must either not leave the source account for an iteration of the recurring transfer, or be transferred back to the source account with no fee.

Decide. Maybe ask on the thread on Slack what people prefer.

Reply:
> Isn't that "if at least one of the keys got capped"? Because that is the condition to have leftover reward balance.

No, that balance could go to other keys that wouldn't otherwise have gotten all of their fees back in rewards. If any key is uncapped you can use it all.

Reply:
Yes the algo needs to be more compelx than the one you outlines. I am not sure the spec needs to be the place to work out the algorithm if the requirement is clear?